apps/web/app/.server/service/proposals.ts

```diff
@@ -67,3 +76,111 @@ export async function nowInSeconds() {
   const block = await l1Rpc.getBlock({ blockTag: "latest" });
   return block.timestamp;
 }
+
+export type ProposalDataResponse = {
+  l2ProposalId: Hex;
+} & (
+  | {
+      ok: true;
+      data: StartUpgradeData;
+      l1ProposalId: Hex;
+      error: null;
+    }
+  | {
+      ok: false;
+      error: string;
+      data: null;
+      l1ProposalId: null;
+    }
+);
+
+async function extractProposalData(txHash: Hex, l2ProposalId: Hex): Promise<ProposalDataResponse> {
+  console.log("txHash", txHash);
+  const receipt = await l2Rpc.getTransactionReceipt({ hash: txHash });
+  const logProof = await fetchLogProof(txHash, 0);
+
+  const l1MessageEventId = "0x3a36e47291f4201faf137fab081d92295bce2d53be2c6ca68ba82c7faa9ce241";
```

Comment:
Can this be deduced from any ABI we have available or from a more human-friendly approach (e.g. the hash of a function selector)? Not particularly against magic variables but we should exhaust other options.